### PR TITLE
【CINN】refactor cuda query component

### PR DESCRIPTION
**Pull request by Courtesy-Xs**
### PR types

Others
### PR changes

Others
### Description

card-72718
This PR remove the ugly code for query nvgpu device info and refactor device info component for cinn

### CODE REVIEW

File: paddle/cinn/common/dev_info_base.h [hunk 1]
```diff
@@ -20,6 +20,10 @@ namespace common {
 class DevInfoBase {
  public:
   virtual ~DevInfoBase() = default;
+  explicit DevInfoBase(int device = 0) : device_num_(device) {}
```
Comment (ZzSean) [inline paddle/cinn/common/dev_info_base.h]: 构造函数和析构函数位置调一下
这里传入的参数名改成`device_num`好一些？

File: paddle/cinn/common/dev_info_manager.cc [hunk 2]
```diff
@@ -19,30 +19,22 @@
...
-#ifdef CINN_WITH_CUDA
-      impl_ = std::make_unique<NVGPUDevInfo>(device_num);
-#endif
-    default:
-      CHECK(false) << "Current device can't be recognized!\n";
-      break;
-  }
+template <Target::Arch arch>
+DevInfoMgr<arch>::DevInfoMgr(int device_num) : device_num_(device_num) {
+  impl = std::make_unique<NVGPUDevInfo>(device_num)
```
Comment (ZzSean) [inline paddle/cinn/common/dev_info_manager.cc]: - impl少个下划线
- 指针类型这里不应该写死吧

Comment (Courtesy-Xs) [reply paddle/cinn/common/dev_info_manager.cc]: done

File: paddle/cinn/common/dev_info_base.h (hunk 1 above)
Comment (Courtesy-Xs) [reply paddle/cinn/common/dev_info_base.h]: done

**Review by ZzSean — APPROVED** [review]

**Pull request merged**